ssl/ssl_text.cc

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 1995-2022 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2005 Nokia. All rights reserved.
+ */
+
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+// Modifications Copyright Amazon.com, Inc. or its affiliates.
+
+
+#include <openssl/ssl.h>
+
+#include "internal.h"
+
+
+int SSL_SESSION_print(BIO *bp, const SSL_SESSION *sess) {
+  if (sess == nullptr) {
+    return 0;
+  }
+
+  bool tls13 = (sess->ssl_version == TLS1_3_VERSION);
+  if (BIO_puts(bp, "SSL-Session:\n") <= 0) {
+    return 0;
+  }
+  if (BIO_printf(bp, "    Protocol  : %s\n", SSL_SESSION_get_version(sess)) <=
+      0) {
+    return 0;
+  }
+
+  if (sess->cipher == nullptr) {
+    if (((sess->cipher->id) & 0xff000000) == 0x02000000) {
```

Comment:
* You verify that `sess->cipher == nullptr` then immediately dereference `sess->cipher->id`?
* Why `0x02000000`? What does that imply about the value?  (The [comment here](https://github.com/aws/aws-lc/blob/4f852cd6f2ef9ee638013636dfb54fa1482d8f62/ssl/ssl_cipher.cc#L1452) says "All OpenSSL cipher IDs are prefaced with 0x03"?)

Reply:
Yeah my bad, I think we can take this out. I took from the original OpenSSL code and hadn't noticed that `sess->cipher->id` was under `sess->cipher`.

All of our available ciphers have names, so the `cipher->id` doesn't really make sense in AWS-LC: https://github.com/aws/aws-lc/blob/4f852cd6f2ef9ee638013636dfb54fa1482d8f62/ssl/ssl_cipher.cc#L158